### ISSUE TITLE
facilitate custom additions to the "preload, create, update" trifecta

### ISSUE THREAD

**Issue by SavedByZero** [photonstorm/phaser#4535]
I write a lot of engines where someone can customize which files are loaded by editing the xml or json file.  The problem is, the rigidity of one preload method makes this hard.  For example, right now I have to do this awkward setup to get my current test to work:
`function preload ()
    {
        this.load.xml('xml', 'assets/test.xml');  
    }

    

    function create ()
    {

         var xmlData = this.cache.xml.get('xml');
        var unit = $(xmlData).find("unit");
        var imageList = $(unit).find("imageList");
        var images = $(imageList)[0].innerHTML.split(",");

        images.forEach(loadImage.bind(this));
          this.load.start();
        this.load.once('complete', finishCreating, this);
        
    }`

I didn't include finishCreating or anything after, but you get the idea.  I would love to be able to just have a more intuitive way to dynamically add methods to the scene object inside of the config object and have them call in order.  (i.e. preload1, preload2, etc.)

**Comment by photonstorm** [comment]
I don't really understand the question here, and this doesn't appear to be an API Bug, more just a general support question - there are much better places to ask those than GitHub.

Regardless, Phaser already has it's own JSON packfile format, which allows you to create a json file that contains all of the game data to be loaded, and then load just this one json file instead. Have a look at the File Pack examples here: http://labs.phaser.io/index.html?dir=loader/&q=

Also, there's no reason why you should only have 'one preload method'. Every Scene can have its own preload method, and you can load files at any point in the Scene, even after preload has finished, or even add files into the load queue as it runs. Again, there are lots of examples in the link above (see the Loader Events folder specifically for this). Scenes also have a files payload config option, which means you could put your XML file in here, and then process its contents at the start of the `preload` method, making whatever loader calls you need.

**Comment by SavedByZero** [comment]
Certainly did not mean to offend with my suggestion. I only made it because, while the information about the kind of feature I need is clearly present as you pointed out, the style of your documentation is frankly a little unclear. All the searches I did kept coming up with three built in functions per state, called in sequence, with no clear explanation of how to add your own custom built in functions to the sequence.
